src/h_assets/cached_file.py

```diff
@@ -0,0 +1,75 @@
+import configparser
+import json
+from os.path import getmtime
+
+from pyramid.settings import aslist
+
+
+class CachedFile:  # pylint:disable=too-few-public-methods
+    """
+    Parses content from a file and caches the result.
+
+    _CachedFile reads a file at a given path and parses the content using a
+    provided loader.
+    """
+
+    path = None
+
+    def __init__(self, path, auto_reload):
+        """
+        Create the CachedFile object.
+
+        :param path: The path to the file to load
+        :param loader: A callable that parses content from a file
+        :param auto_reload: If True, the parsed content is discarded if the
+            mtime of the file changes.
+        """
+
+        self.path = path
+        self._auto_reload = auto_reload
+        self._last_modified_time = None
+        self._cached_content = None
+
+    def load(self):
+        """
+        Return the content of the file parsed with the loader.
+
+        The file is parsed when this is called for the first time and, if
+        auto-reload is enabled, when the mtime of the file changes.
+        """
+
+        current_mtime = getmtime(self.path)
+
+        if not self._cached_content or (
+            self._auto_reload and self._last_modified_time < current_mtime
+        ):
+            with open(self.path) as handle:
+                self._cached_content = self._load_handle(handle)
+
+            self._last_modified_time = current_mtime
```

Comment:
This part was rewritten from before, as it had a hard to see, but unreachable code path in it which was causing coverage problems.

The downside of this approach is it always calls `mtime` although non reloading ones don't technically need it. Pretty sure this is very fast.

Reply:
> The downside of this approach is it always calls mtime although non reloading ones don't technically need it. Pretty sure this is very fast.

IIRC the kernel maintains a cache if data for files that have been recently `stat`-ed, so this is fast for recently accessed files, although there is still some overhead for a syscall. I think this is likely to be fine for our use case, since we're not trying to check thousands of different files.